src/main/resources/static/eiffel-rabbit-mq-post-receive-hook.soy

```diff
@@ -15,25 +15,43 @@
     {param value: $config['rabbit.url'] /}
     {param descriptionText: 'RabbitMQ server address to connect to.' /}
     {param errorTexts: $errors ? $errors['rabbit.url'] : null /}
-    {param rows: '4' /}
+    {param rows: '1' /}
   {/call}
 
   {call aui.form.textareaField}
-    {param id: 'rabbit.exchange' /}
+    {param id: 'rabbit.exchange.name' /}
     {param labelContent: 'RabbitMQ exchange name' /}
-    {param value: $config['rabbit.exchange'] /}
+    {param value: $config['rabbit.exchange.name'] /}
     {param descriptionText: 'RabbitMQ exchange names to send messages to.' /}
-    {param errorTexts: $errors ? $errors['rabbit.exchange'] : null /}
-    {param rows: '4' /}
+    {param errorTexts: $errors ? $errors['rabbit.exchange.name'] : null /}
+    {param rows: '1' /}
   {/call}
 
   {call aui.form.textareaField}
+    {param id: 'rabbit.exchange.type' /}
+    {param labelContent: 'RabbitMQ exchange type' /}
+    {param value: $config['rabbit.exchange.type'] /}
+    {param descriptionText: 'RabbitMQ exchange type.' /}
+    {param errorTexts: $errors ? $errors['rabbit.exchange.type'] : null /}
+    {param rows: '1' /}
+   {/call}
+
+  {call aui.form.textareaField}
     {param id: 'rabbit.routingkey' /}
     {param labelContent: 'RabbitMQ routing key' /}
     {param value: $config['rabbit.routingkey'] /}
     {param descriptionText: 'RabbitMQ routing key to use when sending messages.' /}
     {param errorTexts: $errors ? $errors['rabbit.routingkey'] : null /}
-    {param rows: '4' /}
+    {param rows: '1' /}
+  {/call}
+
+  {call aui.form.textareaField}
+    {param id: 'rabbit.deliverymode' /}
+    {param labelContent: 'RabbitMQ delivery mode' /}
+    {param value: $config['rabbitmq.deliverymode'] /}
+    {param descriptionText: 'RabbitMQ delivery mode. Non-persistent (1) or persistent (2)' /}
+    {param errorTexts: $errors ? $errors['rabbit.deliverymode'] : null /}
```

Comment:
Fails with the following error if left empty

[INFO] [talledLocalContainer] 2016-11-01 10:19:20,847 WARN  [AtlassianEvent::thread-2] admin @CZJWW4x619x609x0 0:0:0:0:0:0:0:1 "POST /scm/project_1/rep_1.git/git-receive-pack HTTP/1.1" c.a.s.i.h.r.AsyncPostReceiveRepositoryHookAdapter Push Event Hook net.praqma.stash.plugins.tracey.hook.EiffelRabbitMQPostReceiveHook failed.
[INFO] [talledLocalContainer] java.lang.NumberFormatException: For input string: ""
[INFO] [talledLocalContainer] 	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[na:1.8.0_65]
[INFO] [talledLocalContainer] 	at java.lang.Integer.parseInt(Integer.java:592) ~[na:1.8.0_65]
[INFO] [talledLocalContainer] 	at java.lang.Integer.parseInt(Integer.java:615) ~[na:1.8.0_65]
[INFO] [talledLocalContainer] 	at com.atlassian.stash.internal.setting.MapSettings.getInt(MapSettings.java:62) ~[MapSettings.class:na]
[INFO] [talledLocalContainer] 	at com.atlassian.stash.internal.setting.MapSettings.getInt(MapSettings.java:68) ~[MapSettings.class:na]
[INFO] [talledLocalContainer] 	at net.praqma.stash.plugins.tracey.components.impl.RabbitMQRoutingInfoConfigService.<init>(RabbitMQRoutingInfoConfigService.java:27) ~[na:na]
[INFO] [talledLocalContainer] 	at net.praqma.stash.plugins.tracey.hook.EiffelRabbitMQPostReceiveHook.postReceive(EiffelRabbitMQPostReceiveHook.java:43) ~[na:na]